### PR TITLE
LITA integration

### PR DESCRIPTION
**Pull request by Slyne**
# What does this PR do ?

This PR aims to add [LITA](https://github.com/NVlabs/LITA) from checkpoint conversion, dataset preprocessing for DVC datasets to finetuning.
It also adds vila to the lita model.

**Collection**: [Note which collection this PR will affect]
* multimodal
* nlp

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by Slyne** [comment]
TO DO LIST:
1. to move `convert_llava_hf_to_nemo.py`
2. to add tutorial

File: examples/multimodal/multimodal_llm/neva/conf/vita_config.yaml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+  # miscellaneous
+  seed: 1234
+  resume_from_checkpoint: null # manually set the checkpoint file to load from
+  apex_transformer_log_level: 30 # Python logging level displays logs with severity greater than or equal to this
+  gradient_as_bucket_view: True # PyTorch DDP argument. Allocate gradients in a contiguous bucket to save memory (less fragmentation and buffer memory)
+
+  tokenizer:
+    library: 'huggingface'
+    type: /ws/converted_models/tokenizer
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/vita_config.yaml:186]: updated to /path/to/vita_tokenizer add a comment how to prepare

File: examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config.yaml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  # miscellaneous
+  seed: 1234
+  resume_from_checkpoint: null # manually set the checkpoint file to load from
+  apex_transformer_log_level: 30 # Python logging level displays logs with severity greater than or equal to this
+  gradient_as_bucket_view: True # PyTorch DDP argument. Allocate gradients in a contiguous bucket to save memory (less fragmentation and buffer memory)
+
+  tokenizer:
+    library: 'sentencepiece'
+    type: null
+    model: /ws/converted_nemo_model/tokenizer_1_5.model
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config.yaml:183]: update

File: examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config_lora.yaml [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  # miscellaneous
+  seed: 1234
+  resume_from_checkpoint: null # manually set the checkpoint file to load from
+  apex_transformer_log_level: 30 # Python logging level displays logs with severity greater than or equal to this
+  gradient_as_bucket_view: True # PyTorch DDP argument. Allocate gradients in a contiguous bucket to save memory (less fragmentation and buffer memory)
+
+  tokenizer:
+    library: 'sentencepiece'
+    type: null
+    model: /ws/converted_nemo_model/tokenizer_1_5.model
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config_lora.yaml:197]: update

File: examples/multimodal/multimodal_llm/neva/conf/lita_config.yaml [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  # miscellaneous
+  seed: 1234
+  resume_from_checkpoint: null # manually set the checkpoint file to load from
+  apex_transformer_log_level: 30 # Python logging level displays logs with severity greater than or equal to this
+  gradient_as_bucket_view: True # PyTorch DDP argument. Allocate gradients in a contiguous bucket to save memory (less fragmentation and buffer memory)
+
+  tokenizer:
+    library: 'sentencepiece'
+    type: null
+    model: null
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/lita_config.yaml:183]: update

File: examples/multimodal/multimodal_llm/neva/eval/eval_qa.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/eval/eval_qa.py:1]: check if image only llava works

File: examples/multimodal/multimodal_llm/neva/eval/eval_rtl.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/eval/eval_rtl.py:1]: add video in script name

File: examples/multimodal/multimodal_llm/neva/neva_evaluation.py [hunk 7]
```diff
@@ -71,34 +156,41 @@ def main(cfg) -> None:
...
+        cfg.get("add_media_sep", False),
     )
+    # num_workers = min(multiprocessing.cpu_count()-1, 4)
+    num_workers = 2
+    dataloader = DataLoader(
+        dataset,
+        batch_size=cfg.inference.get("batch_size", 1),
+        shuffle=False,
+        collate_fn=collate_function,
+        num_workers=num_workers,
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/neva_evaluation.py:191]: persistent_workers=config.num_workers > 0,  # helps Lhotse Shar maintain shuffling state

File: examples/multimodal/multimodal_llm/neva/neva_evaluation.py [hunk 8]
```diff
@@ -71,34 +156,41 @@ def main(cfg) -> None:
...
+    dataset = TemporalNevaDataset(
+        prompt_dicts,
+        cfg.inference.media_base_path,
+        media_token,
+        insert_media_token,
+        image_processor,
+        video_processor,
+        cfg.get("add_media_sep", False),
     )
+    # num_workers = min(multiprocessing.cpu_count()-1, 4)
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/neva_evaluation.py:184]: delete

File: nemo/collections/multimodal/parts/utils.py [hunk 9]
```diff
@@ -449,7 +449,7 @@ def image_processor(maybe_image_path):
     def video_processor(maybe_video_path):
 
         if isinstance(maybe_video_path, str):
-            decord.bridge.set_bridge("torch")
+            # decord.bridge.set_bridge("torch") # this line will break `first`, `middle`, `last`
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/parts/utils.py:452]: delete

File: examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config.yaml [hunk 10]
```diff
@@ -0,0 +1,227 @@
+name: nemo_video_lita_neva
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config.yaml:1]: remove

File: examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config_lora.yaml [hunk 11]
```diff
@@ -0,0 +1,241 @@
+name: nemo_video_lita_lora
```
Comment (yaoyu-33) [inline examples/multimodal/multimodal_llm/neva/conf/lita_1_5_config_lora.yaml:1]: remove

**Review by yaoyu-33 — DISMISSED** [review]

**Review by yaoyu-33 — APPROVED** [review]
